### DISCUSSION TITLE
能否脱离vue使用原生js调用

### DISCUSSION THREAD

**Discussion by passerbyo** [element-plus/element-plus#8527] [Feature Request]
Feature type: **`Other`**

## Details

- Enhanced component name: `All`

## Description

能否脱离vue使用原生js调用

### Additional comments

能否脱离vue使用原生js调用

**Comment by sxzz** [comment]
不能。

**Comment by KimYangOfCat** [comment]
想要 JS 原生调用的话，也许你应该考虑去寻找基于 WebComponents 实现的 UI 库

**Comment by GuoJikun** [comment]
估计是不行，可以找找webComponent组件库

**Comment by ldc2726** [comment]
不能，他本来就是基于vue3所构建的一套UI库，同理vue2有element-ui,它也不能用于vue3的框架，因此，很多UI库都是以框架为前提。

**Comment by heart-to-the-sea** [comment]
不能，不过你可以通过对源码进行修改，将组件进行defineCustomElement进行组件打包，但是需要注意的是其中的外部状态尽量剥离，保证每个组件都是独立的
